teacher.rb

```diff
@@ -0,0 +1,19 @@
+require './person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(age, name, specialization)
+    super(age, name)
```

Comment:
- [ ] It would be better also to initialize the *Teacher* class with `parent_permission` argument and call it as a constructor of the *Person* class 😂 

By calling the constructor of the *Person* class with the `parent_permission` argument included, you ensure that the `parent_permission` attribute is initialized in both the *Teacher* and *Person* classes. This allows you to access the `parent_permission` attribute from both classes and any other subclasses that inherit from *Person*.